cpp/src/partitioning/partitioning.cu

```diff
@@ -770,10 +775,23 @@ std::pair<std::unique_ptr<table>, std::vector<size_type>> hash_partition(
   table_view const& input,
   std::vector<size_type> const& columns_to_hash,
   int num_partitions,
-  rmm::mr::device_memory_resource* mr)
+  hash_id hash_function,
+  rmm::mr::device_memory_resource* mr,
+  cudaStream_t stream)
 {
   CUDF_FUNC_RANGE();
-  return detail::local::hash_partition(input, columns_to_hash, num_partitions, mr);
+
+  switch (hash_function) {
+    case (hash_id::HASH_IDENTITY):
+      return detail::local::hash_partition<IdentityHash>(
+        input, columns_to_hash, num_partitions, mr, stream);
+    case (hash_id::HASH_MURMUR3):
+      return detail::local::hash_partition<MurmurHash3_32>(
+        input, columns_to_hash, num_partitions, mr, stream);
+    default:
+      assert(false && "Unsupported hash function in hash_partition");
+      return std::make_pair(nullptr, std::vector<size_type>());
```

Comment:
```suggestion
      CUDF_FAIL( "Unsupported hash function in hash_partition");
```

Reply:
Agree here. Can we do something similar in device code? The `release_assert` comment is mostly for https://github.com/gaohao95/cudf/blob/6bb99f04551c168ba39c966b113e84fcb3b0d188/cpp/include/cudf/detail/utilities/hash_functions.cuh#L586.

Reply:
No. `release_assert` is the only option in device code. But you should not rely on `release_assert` and should catch any errors on the host beforehand.

Reply:
Implemented in https://github.com/rapidsai/cudf/pull/6726/commits/f9fcc2982a167c61fcb166c84bcd85c050a0e57a.